contracts/globalConstraints/EtherGC.sol

```diff
@@ -0,0 +1,84 @@
+pragma solidity 0.5.17;
+
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "./GlobalConstraintInterface.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+
+/**
+ * @title EtherGC ether constraint per period
+ */
+contract EtherGC {
+    using SafeMath for uint256;
+
+    uint256 public constraintPeriodWindow; //a static window
+    uint256 public amountAllowedPerPeriod;
+    address public avatar;
+    uint256 public latestEthBalance;
+    mapping(uint256=>uint256) public totalAmountSentForPeriods;
+    uint256 public startBlock;
+    address public controller;
+    uint256 public avatarBalanceBefore;
+
+  /**
+   * @dev initialize
+   * @param _avatar the avatar to mint reputation from
+   * @param _constraintPeriodWindow the constraintPeriodWindow in blocks units
+   * @param _amountAllowedPerPeriod the amount of eth to constraint for each period
+   */
+    function initialize(
+        address _avatar,
+        uint256 _constraintPeriodWindow,
+        uint256 _amountAllowedPerPeriod,
+        address _controller
+    )
+    external
+    {
+        require(avatar == address(0), "can be called only one time");
+        require(_avatar != address(0), "avatar cannot be zero");
+        avatar = _avatar;
+        constraintPeriodWindow = _constraintPeriodWindow;
+        amountAllowedPerPeriod = _amountAllowedPerPeriod;
+        latestEthBalance = avatar.balance;
+        startBlock = block.number;
+        controller = _controller;
+    }
+
+    /**
+     * @dev check the constraint before the action.
+     * @return true
+     */
+    function pre(address, bytes32, bytes32) public returns(bool) {
+        require(msg.sender == controller, "only controller is authorize to call");
+        avatarBalanceBefore = avatar.balance;
+        return true;
+    }
+
+    /**
+     * @dev check the allowance of ether sent per period.
+     * @return bool which represents a success
+     */
+    function post(address, bytes32, bytes32) public returns(bool) {
+        require(msg.sender == controller, "only controller is authorize to call");
+        uint256 currentPeriodIndex = (block.number - startBlock)/constraintPeriodWindow;
+        if ((block.number - startBlock) % constraintPeriodWindow > 0) {
+            currentPeriodIndex = currentPeriodIndex.add(1);
+        }
+        if (avatarBalanceBefore >= avatar.balance) {
```

Comment:
so you are checking the netto balance, not how much ether the avatar actually did send (i.e. if the constraint is 1 ETH/day, then it would be ok if the avatar receives 100.000 and sends 99.999)

Reply:
this will not be allowed by this constraint .